Gems/ScriptCanvas/Code/Include/ScriptCanvas/Grammar/AbstractCodeModel.cpp

```diff
@@ -3317,7 +3317,27 @@ namespace ScriptCanvas
 
             // \todo Infinite loop handling both in code and in the graph will have to occur here.
             auto executionOutNodes = execution->GetId().m_node->GetConnectedNodes(outSlot);
-            auto numConnections = executionOutNodes.size();
+
+            // Sort the out node connections to ensure that implicit execution connections are always parsed first
+            EndpointsResolved sortedExecutionOutNodes;
+            for (const EndpointResolved& endpoint : executionOutNodes)
+            {
+                // If this connection is implicit, only parse it if the recieving node has parsed all of its implicit
+                // connections already
+                if (endpoint.second->CreatesImplicitConnections() && !HasUnparsedImplicitConnections(&outSlot, endpoint.second))
+                {
+                    sortedExecutionOutNodes.emplace_back(endpoint);
+                }
+            }
+            for (const EndpointResolved& endpoint : executionOutNodes)
+            {
+                if (!endpoint.second->CreatesImplicitConnections())
+                {
+                    sortedExecutionOutNodes.emplace_back(endpoint);
+                }
+            }
```

Comment:
If you are just grouping to make sure the parsed implicit nodes are done first, you can probably optimize this to only loop through once by emplace_front and emplace_back
```suggestion
            for (const EndpointResolved& endpoint : executionOutNodes)
            {
                // If this connection is implicit, only parse it if the recieving node has parsed all of its implicit
                // connections already
                if (endpoint.second->CreatesImplicitConnections() && !HasUnparsedImplicitConnections(&outSlot, endpoint.second))
                {
                    sortedExecutionOutNodes.emplace_front(endpoint);
                }
                else if (!endpoint.second->CreatesImplicitConnections())
                {
                    sortedExecutionOutNodes.emplace_back(endpoint);
                }
            }
```

Reply:
Behind the scenes, EndpointsResolved seems to be a vector which means that it doesn't have an emplace_front method. I could insert the items into the front of the list, but each time I do that the vector would have to shift all other elements which would hurt efficiency https://www.geeksforgeeks.org/vector-insert-function-in-cpp-stl/. Not a huge deal though, since I don't expect this vector to consist of any more than 2-3 elements in most scenarios. But I do need the executionOutNodes with the implicit connections added to that vector in that order (first in first out) for parsing to work with implicitly connected nodes that are chained together.

Reply:
Yeah, its really just a minor observation and not worth it to over analyze (I didn't catch that it was a vector). But the shift isn't a big deal in this case anyways since its a vector of references. Leaving your original solution will at least preserve the order of like items anyways, if that was important at all